resources/matrix_irc.rb

```diff
@@ -0,0 +1,121 @@
+# osl_matrix_irc
+#
+# Matrix-Appservice-IRC is an IRC bouncer developed by Matrix.
+
+resource_name :osl_matrix_irc
+provides :osl_matrix_irc
+unified_mode true
+
+default_action :create
+
+property :container_name, String, name_property: true
+property :config, Hash, default: {}
+property :port, Integer, default: 8090
+property :host_domain, String, required: true
+property :host_name, String, default: lazy { "synapse-#{host_domain}" }
+property :host_path, String, default: lazy { "/opt/#{host_name}" }
+property :key_appservice, String, default: lazy { osl_matrix_genkey(host_name + container_name) }
+property :key_homeserver, String, default: lazy { osl_matrix_genkey(container_name + host_name) }
+property :tag, String, default: 'latest'
+property :sensitive, [true, false], default: true
+
+action :create do
+  # Pull down the latest version
+  docker_image 'matrixdotorg/matrix-appservice-irc' do
+    tag new_resource.tag
+  end
```

Comment:
When this image gets updated, how does the docker compose know that it needs to be updated? Have we tested this yet to make sure it works?

I believe in other cookbooks I've notified restarts when this gets triggered.

Reply:
In the context of us running a Matrix server manage by `osl_synapse_service`, which we use in `osl-nodes`. As these appservices have been abstracted by being their own resource. When `docker_image` pulls a new image down, it should also trigger the parent `osl_matrix_irc`, which then [notifies docker to rebuild within `osl_synapse_service`](https://github.com/osuosl-cookbooks/osl-matrix/blob/zwall/matrix-appservice-irc/resources/synapse_service.rb#L77)

Reply:
I've double checked this feature, and can confirm that `osl_dockercompose`'s `rebuild` action will successfully keep all `latest` tagged containers updated.